test/configparser.cpp

```diff
@@ -106,17 +106,21 @@ TEST_CASE("include directive includes other config files", "[ConfigParser]")
 	SECTION("Diamond of death includes pass") {
 		REQUIRE_NOTHROW(cfgparser.parse("data/diamond-of-death/A"));
 	}
-	SECTION("Recursive include don't happen") {
+	SECTION("File including itself only gets evaluated once") {
 		TestHelpers::TempFile testfile;
-		TestHelpers::EnvVar("$TMPFILE").set(testfile.getPath()); // $TMPFILE used in conf file
+		TestHelpers::EnvVar tmpfile("TMPFILE"); // $TMPFILE used in conf file
+		tmpfile.set(testfile.getPath());
 
 		REQUIRE_NOTHROW(cfgparser.parse("data/recursive-include-side-effect")); // recursive includes don't fail
 		// I think it will never get below here and fail? If it recurses, the above fails
 
-		std::ifstream in(testfile.getPath(), std::ifstream::ate | std::ifstream::binary);
-		REQUIRE(in.tellg() < 100); // 100 is just a reasonably high number for one command output
-
-		// TODO: timeout somehow? wouldn't want the tests going on forever if it
-		// actually is recursing
+		int line_count = 0;
+		{ // from https://stackoverflow.com/a/19140230
+			std::ifstream in(testfile.getPath());
+			std::string line;
+			while (std::getline(in, line))
```

Comment:
Please always use curly braces around blocks, in case they're extended with more statements later.

(We tried to offload comments like this to `clang-format`, just like we did with `rustfmt` for Rust, but there are multiple major versions of `clang-format` in use, and they produce different results. We gave up for now, so I'm writing these nitpicks by hand. See #198 for more details on clang-format situation.)